### ISSUE TITLE
Build URL is not captured on TeamCity

### ISSUE THREAD

**Issue by oehme** [gradle/gradle-enterprise-build-config-samples#75]
The code currently looks for a variable named `BUILD_URL` on TeamCity. This variable does not exist on TeamCity. You would have to compute the URL some other way.

This probably originated from Gradle's own build - There is a variable named `BUILD_URL` there, but it's custom-configured, not a standard variable that JetBrains provides.

**Comment by etiennestuder** [comment]
We used to do this differently, but then the code was removed recently. I was pretty sure there was a reason why the previous code was more complex than now :-)

//cc @bigdaz @runningcode @facewindu

**Comment by oehme** [comment]
Just looked at the commit and it refers to TeamCity 2020.2 - The server in question is on 2020.1, so that would explain it. Maybe just add some validation that the TeamCity version is as expected?

**Comment by bigdaz** [comment]
I could have sworn that I only used documented environment variables for TeamCity, but now I look I can find `BUILD_NUMBER` but not `BUILD_URL` or `BUILD_AGENT_NAME`.

I guess the only way to test this properly would be on a vanilla TeamCity install. We should also test against a few different versions if possible.

**Comment by bigdaz** [comment]
This is the original code, which wasn't working our our internal server, but might serve as a basis for the next iteration:
https://github.com/gradle/gradle-enterprise-build-config-samples/blob/common-custom-user-data-maven-extension-1.2/common-custom-user-data-scripts/gradle-common-custom-user-data.gradle#L85-L104

**Comment by bigdaz** [comment]
I did some investigation with a vanilla local installation of TeamCity 2020.2.2, and here's what I found:
- Environment variables: The few predefined Environment variables that are documented are available as Environment variables in the executing build. 
    - Several undocumented Environment variables are present as well
- System properties: None of the predefined System properties are available as System properties in the executing build.
    - However, these predefined System properties _are_ available as _project_ properties in the executing build. While surprising to me, this is the [documented behaviour](https://www.jetbrains.com/help/teamcity/configuring-build-parameters.html#Using+Build+Parameters+in+the+Build+Scripts). 

Here's a build scan that list all environment variables, system properties and project properties for a build invoked by TeamCity. https://e.grdev.net/s/24emmmg6tzwoe/custom-values

With this information, I'm sure we can come up with something that works with TeamCity 2020.2. Next will be to see if this behaviour is different in 2020.1 and earlier versions.
